### DISCUSSION TITLE
Why does FSP report an error you must have exactly 2 fields？

### DISCUSSION THREAD

**Discussion by nanguaxiaofendui** [idaholab/moose#23695] [Q&A General] [answered]
Hello, everyone!
When I used FSP preprocessing, the following error occurred. How can I correct it? Thank you very much！
![1678626434230(1)](https://user-images.githubusercontent.com/126685008/224546624-384ea0ba-3443-4f5b-a764-54fd17d486b7.png)
![1678626479352(1)](https://user-images.githubusercontent.com/126685008/224546665-763c8099-6c02-42ea-9cbf-213e361cc75d.png)

**Comment by GiudGiud** [comment, marked as answer]
Hello

The Schur decomposition is well defined for two groups of variable, but has not been extended to more than that

if you want more splits than two groups then you need to use a multipliciative or additive field split instead of Schur.
They may not converge as well from our experience

Guillaume
